crypto/params.c

```diff
@@ -13,6 +13,14 @@
 #include "internal/thread_once.h"
 #include "internal/numbers.h"
 
+/*
+ * Helper macros that know the number of mantassia bits in floats and doubles.
+ * Doubles have 53 bit mantissas (52 stored, 1 implicit).
+ * Floats have 24 bit mantissas (23 stored, 1 implicit).
```

Comment:
Are you sure this is universally true?

Reply:
They should be on the conservative side.  The popular machines I remember where it wasn't were some IBM mainframes and the VAX.  The VAX has the F format which does have 23 mantissa bits but the D format has 56 rather than 53.  IBM used 24 and 55 bits and rounded in hexadecimal rather than binary :weary:.  Early Microsoft BASICs used 23 and 55 bits for its 32 and 64 bit formats.  Of these, I think that the only one that is currently supported by OpenSSL is the S390 by IBM.

IEEE 754 caught on very rapidly and very widely.  It was a really good standard.

There have been a [myriad of oddities](http://www.quadibloc.com/comp/cp0201.htm).  Microsoft BASIC has a 40 bit format, control data used 60 bit reals, Cray has a 64 bit real, etc.

Reply:
I think I can come up with an improved alternative.